libraries/azure_backend.rb

```diff
@@ -236,7 +236,7 @@ def get_resource(opts = {})
                                 allow: %i(api_version),
                                 opts: opts)
     api_version = opts[:api_version] || 'latest'
-    if opts[:resource_uri].include?('providers')
+    if opts[:resource_uri].scan('providers').size == 1
```

Comment:
would `.match?` be sufficient here if we are only needing one instance of `providers`?
```
if opts[:resource_uri].match?('providers')
```